lib/src/command/barback.dart

```diff
@@ -24,6 +31,12 @@ abstract class BarbackCommand extends PubCommand {
   /// The build mode.
   BarbackMode get mode => new BarbackMode(argResults["mode"]);
 
+  // The current compiler mode.
+  CompilerMode get compilerMode =>
+      argResults.options.contains('dart2js') && argResults.wasParsed('dart2js')
+          ? argResults['dart2js'] ? CompilerMode.dart2Js : CompilerMode.none
+          : _compilerArgToMode[argResults["compiler"]];
```

Comment:
A nested conditional operator is not exactly reader friendly. :)

I think this chunk of code is big enough to break into a statement or two.